tsptw_data_dt.h

```diff
@@ -1054,15 +1114,57 @@ void TSPTWDataDT::LoadInstance(const std::string& filename) {
   // Compute horizon
   horizon_     = 0;
   max_service_ = 0;
+  max_rest_    = 0;
+  int64 rest_duration;
+  for (int32 v = 0; v < tsptw_vehicles_.size(); v++) {
+    rest_duration = 0;
+    for (int32 r = 0; r < tsptw_vehicles_[v].Rests().size(); r++) {
+      rest_duration += tsptw_vehicles_[v].Rests()[r].duration;
+    }
+    max_rest_ = std::max(max_rest_, rest_duration);
+  }
+  if (AllVehiclesHaveEnd()) {
+    for (int32 v = 0; v < tsptw_vehicles_.size(); v++) {
+      horizon_ = std::max(horizon_, tsptw_vehicles_[v].time_end +
+                                        tsptw_vehicles_[v].time_maximum_lateness);
+    }
+  } else if (AllServicesHaveEnd()) {
+    for (int32 i = 0; i < tsptw_clients_.size(); i++) {
+      horizon_ = std::max(horizon_, tsptw_clients_[i].due_time.back() +
+                                        tsptw_clients_[i].maximum_lateness.back());
+    }
+    for (int32 v = 0; v < tsptw_vehicles_.size(); v++) {
+      for (int32 r = 0; r < tsptw_vehicles_[v].Rests().size(); r++) {
+        horizon_ = std::max(horizon_, tsptw_vehicles_[v].Rests()[r].due_time);
+      }
+    }
+  } else {
+    int64 latest_start    = 0;
+    int64 latest_rest_end = 0;
+    int64 sum_service     = 0;
+    int64 sum_setup       = 0;
+    for (int32 i = 0; i < size_missions_; ++i) {
+      sum_service += tsptw_clients_[i].service_time;
+      sum_setup += tsptw_clients_[i].setup_time;
+      if (tsptw_clients_[i].ready_time.size() > 0) {
+        latest_start = std::max(latest_start, tsptw_clients_[i].ready_time.back());
+      }
+    }
+    for (int32 v = 0; v < tsptw_vehicles_.size(); v++) {
+      latest_start = std::max(latest_start, tsptw_vehicles_[v].time_start);
+
+      for (int32 r = 0; r < tsptw_vehicles_[v].Rests().size(); r++) {
+        latest_rest_end =
+            std::max(latest_rest_end, tsptw_vehicles_[v].Rests()[r].due_time);
+      }
+    }
+    horizon_ = std::max(latest_start, latest_rest_end) + sum_service * max_coef_service_ +
+               sum_setup * max_coef_setup_ + sum_max_time_ + sum_lapse + max_rest_;
+  }
+
   for (int32 i = 0; i < size_missions_; ++i) {
-    if (tsptw_clients_[i].due_time.size() > 0)
-      horizon_ = std::max(horizon_, tsptw_clients_[i].due_time.back());
     max_service_ = std::max(max_service_, tsptw_clients_[i].service_time);
   }
```

Comment:
If `max_service_` is not use anywhere, let remove it.

Reply:
It is currently used in tsp_simple.cc through MaxServiceTime()